R/generateSettings.R

```diff
@@ -30,26 +31,38 @@
 #' 
 #' @export
 
-generateSettings <- function(standard="None", chart="eDish", partial=FALSE, partial_cols=NULL){
+generateSettings <- function(standard="None", chart="eDish", requiredOnly=TRUE, partial=FALSE, partial_keys=NULL){
   if(tolower(chart)!="edish"){
     stop(paste0("Can't generate settings for the specified chart ('",chart,"'). Only the 'eDish' chart is supported for now."))
   }
   
-  # Check that partial_cols is supplied if partial is true
-  if  (is.null(partial_cols) & partial ) {
-    stop("partial_cols must be supplied if the standard is partial")
+  # Check that partial_keys is supplied if partial is true
+  if  (is.null(partial_keys) & partial ) {
+    stop("partial_keys must be supplied if the standard is partial")
   }
   
-  metadata <- safetyGraphics::getSettingsMetadata(
-    charts = chart, 
-    cols=c("text_key","adam","sdtm"),
-    filter_expr = .data$adam != '' & .data$sdtm != '' 
-  )
-
-  # Split on -- for multi-level handling 
-  hierarchical_metadata <- str_split(metadata$text_key, "--") 
+  # Coerce options to lowercase
+  standard<-tolower(standard)
+  chart<-tolower(chart)
+  
+  # Build a table of data mappings for the selected standard and partial settings
+  if(standard != "none"){
+    dataMappings <- safetyGraphics::getSettingsMetadata(
+      charts = chart, 
+      cols=c("text_key",standard,"setting_required")
+    ) %>% 
+    filter(ifelse(requiredOnly, .data$setting_required, TRUE))%>%
+    rename("column_name" = standard)%>%
+    filter(.data$column_name != '')
+    
+    if(partial){
+      dataMappings<-dataMappings%>%filter(.data$text_key %in% partial_keys) 
+    }
+  }
   
-  settings<-list(
+  # build shell settings for each chart (move these to /data eventually?)
+  shells<-list()
```

Comment:
and maybe you only need to write the shell that matches the chart setting (so don't have to make a big list of them) - but that's not really a concern right now

Reply:
Filed #168 to cover this work.